src/controllers/call_controller.cr

```diff
@@ -0,0 +1,122 @@
+require "json"
+require "./controller"
+require "../db"
+require "../lib/bad_request_exception"
+require "../models/reply_command"
+require "../models/ao_message"
+require "../models/respondent"
+
+class Twiliosim::CallController < Twiliosim::Controller
+  def initialize(context : HTTP::Server::Context, @account_sid : String)
+    @db = Twiliosim::DB.get_instance()
+    super(context)
+  end
+
+  def handle_request
+    body_params = get_call_request_body_params(@context.request)
+    from = body_params["from"]
+    to = body_params["to"]
+    verboice_url = body_params["verboice_url"]
+
+    call = create_and_start_call(to, from, @account_sid)
+    response_call_created(@context, call.id)
+    spawn do
+      # We give Verboice a sec to process the response. Just in case it needs it.
+      sleep 1.seconds
+
+      # verboice_url cannot be nil, so the `not_nil!` call shouldn't be here.
+      # It's included here just because of [this Crystal compiler known issue](https://github.com/crystal-lang/crystal/issues/3093)
+      handle_created_call(verboice_url.not_nil!, call)
+    end
+  end
+
+  private def create_and_start_call(to : String, from : String, @account_sid : String) : TwilioCall
+    call = @db.create_call(to, from, account_sid)
+    call.start()
+    @db.update_call(call)
+  end
+
+  private def finish_and_update_call(call : TwilioCall) : TwilioCall
+    call.finish()
+    @db.update_call(call)
+  end
+
+  private def get_call_request_body_params(request : HTTP::Request) : {to: String, from: String, verboice_url: String}
+    params = get_validated_body_params(request, ["From", "To", "Url"])
```

Comment:
the arguments can be a `Tuple{String, String, String}` and recevied as `Enumerable(String)`. This will avoid allocating an array for each request.